pages/agent/v3/hooks.md

```diff
@@ -1,102 +1,111 @@
+Hooks allow you extend or override the built-in behaviour of the Buildkite Agent as it runs jobs in a variety of phases using a variety of programming languages.
+
+### What's a hook?
+
+At its core, a hook is a script that is executed or sourced by the Buildkite agent at a specific point in the job lifecycle. Hooks can be used to extend or override the built-in behaviour of the Buildkite agent. Hooks are generally run as shell scripts, which the agent then executes or sources.
+
+>🛠 Experimental feature
+> Using the `polyglot-hooks` experiment, agents running v3.47.0 or later can run hooks written in any programming language. See the [polyglot hooks](#polyglot-hooks) section for more information.
+
+### Hook scopes
 
 Hooks can be defined in the following locations:
 
-* On the agent
-* In your pipeline's repository
+* In the filesystem of the agent machine (called _agent hooks_, or more rarely _global hooks_)
+* In your pipeline's repository (called _repository hooks_, or more rarely _local hooks_)
 * In [plugins](/docs/plugins) applied to steps
 
-Hooks defined on the agent are also referred to as _global hooks_, while hooks defined in the pipeline's repository are known as _local hooks_.
-
-For example, you could define an agent-wide `checkout` hook which spins up a fresh `git clone` on a new build
-machine, a repository `pre-command` hook which sets up repository-specific
-environment variables, or a plugin `environment` hook which fetches API keys
-from a secrets' storage service.
+For example, you could define an agent-wide `checkout` hook which spins up a fresh `git clone` on a new build machine, a repository `pre-command` hook which sets up repository-specific environment variables, or a plugin `environment` hook which fetches API keys from a secrets' storage service.
 
 There are two categories of hook:
 
 * Agent Lifecycle
 * Job Lifecycle
 
-Agent lifecycle hooks are **executed** by the Buildkite agent as part of the
-agent's lifecycle. For example, the `pre-bootstrap` hook is executed before
-starting a job's bootstrap process, and the `agent-shutdown` hook is executed
-before the agent process terminates.
+Agent lifecycle hooks are **executed** by the Buildkite agent as part of the agent's lifecycle. For example, the `pre-bootstrap` hook is executed before starting a job's bootstrap process, and the `agent-shutdown` hook is executed before the agent process terminates.
+
+Job lifecycle hooks are **sourced** (kinda sorta, see "📝 A note on sourcing for shellscripting nerds" below) by the Buildkite bootstrap in the different job phases. They are run in a per-job shell environment, and any exported environment variables are carried to the job's subsequent phases and hooks. For example, the `environment` hook can modify or export new environment variables for the job's subsequent checkout and command phases. Shell options set by individual hooks, such as set `-e -o pipefail`, are not carried over to other phases or hooks.
```

Comment:
> … (kinda sorta, see “…

IMO “kinda sorta” goes a bit beyond Buildkite's casual voice, and could be deleted without losing meaning, but I'm not violently opposed to it :)